runner_test.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/consul-template/config"
+	"github.com/hashicorp/consul-template/dependency"
+)
+
+func TestRunner_appendSecrets(t *testing.T) {
+	t.Parallel()
+
+	secretValue := "somevalue"
+
+	cases := map[string]struct {
+		path string
+		data *dependency.Secret
+		err  bool
+	}{
+		"kv1_secret": {
+			"kv/bar",
+			&dependency.Secret{
+				Data: map[string]interface{}{
+					"key_field": secretValue,
+				},
+			},
+			false,
+		},
+		"kv2_secret": {
+			"secret/data/foo",
+			&dependency.Secret{
+				Data: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"destroyed": bool(false),
```

Comment:
Should we be checking this field during the parsing above? I couldn't find a description in the vault docs of whether the secret would still be populated when this field is true.

Reply:
If the secret is deleted or destroyed, it wont be returned. If it's delete, you will get a response with `metadata` but `data` will be nil:

```json
[vault][master](4)$ curl -s --header "X-Vault-Token: $VAULT_TOKEN" http://127.0.0.1:8200/v1/kv/data/something | jq                                                                                                                                           ✭
{
  "request_id": "6e926af4-37a9-2e17-679f-e9d7c7553485",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": {
    "data": null,
    "metadata": {
      "created_time": "2018-11-14T22:39:13.348641Z",
      "deletion_time": "2018-11-14T22:43:55.199587Z",
      "destroyed": false,
      "version": 2
    }
  },
  "wrap_info": null,
  "warnings": null,
  "auth": null
}
```

In this case, we enter the `default:` case and no secret is populated  (https://github.com/hashicorp/envconsul/pull/186/files#diff-a0afb4d13dc79c28cad59d370ba902caR451)

Reply:
@kyhavlov I added / edited the tests in 3884fc8c99ce927acc138812824c7ff0ecae4550 , I believe that covers it but please take another look when you can